src/bin/writedisk.rs

```diff
@@ -95,6 +95,21 @@ impl UsbBlockDevice {
     }
 }
 
+fn is_device_mounted(device: &UsbBlockDevice) -> bool {
+    // Device chosen by the user.
+    let chosen_device = device.device.to_str().unwrap();
+    // Get a list of all currently mounted devices/partitions.
+    let mounted_devices: Vec<String> = procfs::mounts()
+        .unwrap()
+        .into_iter()
+        .map(|x| x.fs_spec)
+        .collect();
+    // Return whether or not the user's chosen device is currently mounted.
+    mounted_devices
+        .iter()
+        .any(|dev| dev.starts_with(chosen_device))
```

Comment:
These two iterations can be combined, something like:
```rust
    procfs::mounts()
        .unwrap()
        .iter()
        .any(|x| x.fs_spec.starts_with(chosen_device))
```

Reply:
That's much simpler, I have trouble wrapping my brain around the chaining. Implemented in [034904c](https://github.com/nicholasbishop/writedisk/pull/277/commits/034904cd09d746656b130b9bf280a3e8d132f423)